ansible/roles/sysprep/tasks/redhat.yml

```diff
@@ -70,3 +70,5 @@
     - lookup('env', 'RHSM_USER') | length > 0
     - lookup('env', 'RHSM_PASS') | length > 0
     - ansible_distribution == 'RedHat'
+    - "packer_builder_type not in ['amazon']"
```

Comment:
removing redhat repositories are part of the `Remove subscription` block. This requires that system is registered with redhat and redhat repos are enabled using subscription manager. With RHEL 8.6 subscription manager throws error if repos are not subscribed before. We only need to subscribe to RHEL for vsphere non-airgapped builds only.

based on testing I will add GCP and Azure to the list.